lib/bridgetown-inline-svg/markup.rb

```diff
@@ -0,0 +1,96 @@
+# Converts the mark passed to the tag into a path & a hash of arguments.
+module BridgetownInlineSvg
+  class Markup
+    attr_reader :markup
+
+    # Separate file path from other attributes
+    PATH_SYNTAX = %r{
+      ^(?<path>[^\s"']+|"[^"]+"|'[^']+')
+      (?<params>.*)
+    }x
+
+    # Parse the first parameter in a string, giving :
+    #  [full_match, param_name, double_quoted_val, single_quoted_val, unquoted_val]
+    # The Regex works like :
+    # - first group
+    #    - match a group of characters that is alphanumeric, _ or -.
+    # - second group (non-capturing OR)
+    #    - match a double-quoted string
+    #    - match a single-quoted string
+    #    - match an unquoted string matching the set : [\w\.\-#]
+    #
+    PARAM_SYNTAX = %r{
+    ([\w-]+)\s*=\s*
+      (?:"([^"\\]*(?:\\.[^"\\]*)*)"|'([^'\\]*(?:\\.[^'\\]*)*)'|([\w.\-#]+))
+    }x
```

Comment:
I'd love to figure if we can remove these regexes & use something closer to what Liquid uses ( https://github.com/Shopify/liquid/blob/57c9cf64ebc777fe5e92d4408d31a911f087eeb4/lib/liquid/tags/render.rb#L27 ).

Reply:
Definitely a refactoring I'd accept but it works for now :smile: